dspace-api/src/main/java/org/dspace/versioning/DefaultItemVersionProvider.java

```diff
@@ -80,8 +80,8 @@ public void deleteVersionedItem(Context c, Version versionToDelete, VersionHisto
                 // version gets archived.
                 Item item = versionHistoryService.getPrevious(c, history, versionToDelete).getItem();
                 if (!item.isArchived()
-                        || workspaceItemService.findByItem(c, versionToDelete.getItem()) != null
-                        || workflowItemService.findByItem(c, versionToDelete.getItem()) != null)
+                        && workspaceItemService.findByItem(c, versionToDelete.getItem()) == null
+                        && workflowItemService.findByItem(c, versionToDelete.getItem()) == null)
```

Comment:
I wonder why the test is not (previous item is not archived and item to be deleted is archived).  An item which is not in a workspace, and not in a workflow, and not archived, should represent an inconsistency in the model -- every item should be in one of these three states, yes?